Lib/dataclasses.py

```diff
@@ -1173,7 +1182,11 @@ class C:
             continue
 
         if f.name not in changes:
-            changes[f.name] = getattr(obj, f.name)
+            if f._field_type is _FIELD_INITVAR:
+                raise ValueError(f"{f._field_type} '{f.name}' "
```

Comment:
For consistency with the rest of the code, please change `'{f.name}'` to `{f.name!r}`. Also, just hard-code `InitVar` in the message, since this message is only applicable with `InitVar`s.